src/circuit/gadget/sinsemilla/chip/generator_table.rs

```diff
@@ -35,7 +35,7 @@ impl GeneratorTableConfig {
 
         meta.lookup(|meta| {
             let q_s1 = meta.query_selector(config.q_sinsemilla1);
-            let q_s2 = meta.query_fixed(config.q_sinsemilla2, Rotation::cur());
+            let q_s2 = meta.query_selector(config.q_sinsemilla2);
             let q_s3 = {
```

Comment:
This is still deriving a composite value from `q_sinsemilla2`, where it should instead be querying `q_sinsemilla3`.

Reply:
it works because: since `q_s2` is always 0 or 1 now, `q_s3 = (q_s2) * (1 - q_s2)` is always 0. `q_s3` is used in the expression 
```
m_{i+1} = z_{i} - 2^K * (q_s2 - q_s3) * z_{i + 1}
```
so that reduces to `m_{i+1} = z_{i} - 2^K * (q_s2) * z_{i + 1}` which is actually the correct expression after the selector updates.

Reply:
(of course i'm going to just remove `q_s3` though)